From: Michael Biebl <biebl@debian.org>
Date: Wed, 20 Aug 2014 23:52:47 +0200
Subject: Stop syslog.socket when entering emergency mode

When rsyslog is installed, the syslog.socket is active when entering emergency
mode. Any log message then triggers the start of rsyslog.service (or other
implementation) along with its dependencies such as local-fs.target and
sysinit.target. As these might fail themselves (e. g. faulty /etc/fstab), this
breaks the emergency mode.

This causes syslog.socket to fail with "Failed to queue service startup job:
Transition is destructive".

Add Conflicts=syslog.socket to emergency.service to make sure the socket is
stopped when emergency.service is started.

(This is a consequence of Re-enable-journal-forwarding-to-syslog.patch)

Closes: #755581
Forwarded: https://github.com/systemd/systemd/issues/266
---
 units/emergency.service.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/units/emergency.service.in b/units/emergency.service.in
index 8dc3cbd..5f7f5e7 100644
--- a/units/emergency.service.in
+++ b/units/emergency.service.in
@@ -11,6 +11,7 @@ Documentation=man:sulogin(8)
 DefaultDependencies=no
 Conflicts=shutdown.target
 Conflicts=rescue.service
+Conflicts=syslog.socket
 Before=shutdown.target
 
 [Service]
